### PR TITLE
feat: update VKs and bump cargo.lock

### PR DESCRIPTION
**Pull request by Artemka374**
## What ❔

update VKs and bump cargo.lock

## Why ❔

for protocol upgrade

## Checklist

- [ ] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [ ] Tests for the changes have been added / updated.
- [ ] Documentation comments have been added / updated.
- [ ] Code has been formatted via `zk fmt` and `zk lint`.
- [ ] Spellcheck has been run via `zk spellcheck`.

### CODE REVIEW

**Review by StanislavBreadless — DISMISSED** [review]

File: Cargo.lock [hunk 1]
```diff
@@ -1533,7 +1533,7 @@ dependencies = [
 [[package]]
 name = "cs_derive"
 version = "0.1.0"
-source = "git+https://github.com/matter-labs/era-boojum.git?branch=main#cd631c9a1d61ec21d7bd22eb74949d43ecfad0fd"
+source = "git+https://github.com/matter-labs/era-boojum.git?branch=main#4bcb11f0610302110ae8109af01d5b652191b2f6"
```
Comment (perekopskiy) [inline Cargo.lock:1536]: please revert cs_derive, it's a part of boojum

File: prover/Cargo.lock [hunk 2]
```diff
@@ -1435,7 +1435,7 @@ dependencies = [
 [[package]]
 name = "cs_derive"
 version = "0.1.0"
-source = "git+https://github.com/matter-labs/era-boojum?branch=main#cd631c9a1d61ec21d7bd22eb74949d43ecfad0fd"
+source = "git+https://github.com/matter-labs/era-boojum?branch=main#4bcb11f0610302110ae8109af01d5b652191b2f6"
```
Comment (perekopskiy) [inline prover/Cargo.lock:1438]: same

**Review by perekopskiy — DISMISSED** [review]

File: prover/vk_setup_data_generator_server_fri/src/keystore.rs [hunk 3]
```diff
@@ -44,7 +44,18 @@ pub struct Keystore {
...
 fn get_base_path() -> PathBuf {
-    core_workspace_dir_or_current_dir().join("prover/vk_setup_data_generator_server_fri/data")
+    let path = core_workspace_dir_or_current_dir();
+
+    let new_path = path.join("prover/vk_setup_data_generator_server_fri/data");
+    if new_path.exists() {
+        return new_path;
+    }
+
+    let mut components = path.components();
```
Comment (mm-zk) [inline prover/vk_setup_data_generator_server_fri/src/keystore.rs:54]: please add a comment what is happening here.

Comment (perekopskiy) [reply prover/vk_setup_data_generator_server_fri/src/keystore.rs:54]: This is a kludge for setup data generator to work in CI. Unfortunately I fail to understand how it's supposed to work, it seems in production it also doesn't work as expected and relies on the fact that we run binaries from some path.

**Review by perekopskiy — COMMENTED** [review]
please also update contracts to include changes to `Verifier`

**Review by perekopskiy — APPROVED** [review]

**Pull request merged**